silver/models/transactions/transaction.py

```diff
@@ -136,6 +136,43 @@ def refund(self, refund_code='default', refund_reason='Unknown refund reason'):
         self.refund_code = refund_code
         logger.error(str(refund_reason))
 
+    def save(self, *args, **kwargs):
+        previous_instance = get_object_or_None(Transaction, pk=self.pk) if self.pk else None
+        setattr(self, 'previous_instance', previous_instance)
+
+        def _clean_and_save():
+            if not getattr(self, '.cleaned', False):
+                self.full_clean(previous_instance=previous_instance)
+
+            super(Transaction, self).save(*args, **kwargs)
+
+        if not previous_instance:
+            # Creating a new Transaction, therefore a lock is used to ensure there are no
+            # overcharges
+            if self.proforma:
+                lock_key = 'proforma_{pk}'.format(pk=self.proforma.pk)
+            elif self.invoice:
+                lock_key = 'invoice_{pk}'.format(pk=self.invoice.pk)
+            else:
+                raise ValidationError(
+                    'The transaction must have at least one document '
+                    '(invoice or proforma).'
+                )
+
+            lock = redis.lock(lock_key, timeout=5)  # 5 seconds should be more than enough to save
+
+            if not lock.acquire(blocking_timeout=5):
+                raise ValidationError("Couldn't create a transaction for the given document.")
```

Comment:
It doesn't look like a `ValidationError`.
Also, maybe some details regarding why the transaction can't be created will help debugging.